lucene/core/src/java/org/apache/lucene/search/LRUQueryCache.java

```diff
@@ -262,6 +283,11 @@ DocIdSet get(Query key, IndexReader.CacheHelper cacheHelper) {
     assert lock.isHeldByCurrentThread();
     assert key instanceof BoostQuery == false;
     assert key instanceof ConstantScoreQuery == false;
+
+    if (inFlightAsyncLoadQueries.contains(key)) {
```

Comment:
Can you add to javadocs for this method explaining when/why this exception is thrown?